Add crust to build storage (#2986)
w3f/polkadot-wiki: docs/build/build-storage.md

This can be viewed as the shift from the centralized *location-centric* storage approach to the 
decentralized *content-centric* approach.

> ### Crust

Reviewer: ```suggestion
### Crust Network
```

> ### Crust

Reviewer: As Crust is built on top of Substrate, can we add this under the Substrate Storage heading, after the content already included?

As in 

```
## Substrate Storage
...
...
### Crust Storage
```

Reviewer: Also needs prettier

Author: Yeah, I think it's reasonable!

> [Crust](https://crust.network) provides a Web3.0 decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy and assurance. Crust supports multiple storage-layer protocols such as IPFS, and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.

Reviewer: ```suggestion
[Crust Network](https://crust.network) provides a decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy, and assurance. Crust supports multiple storage-layer protocols such as [IPFS](#ipfs-interplanetary-file-system) and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.
```

> The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) procotol, it matches people who have hard drive space to spare with those users that need to store data or host content.

Reviewer: ```suggestion
```

Reviewer: Mention of IPFS earlier on which implies the following, and should be enough for the summary

> Crust builds on top of Substrate and provides a native cross-chain communnication pallet based on [XCMP](https://wiki.polkadot.network/docs/learn-crosschain) called [xStorage](https://github.com/crustio/crust/tree/parachain/shadow/crust-collator/pallets/xstorage). The basic idea of xStorage shows below:

Reviewer: ```suggestion
Crust provides a native cross-chain communication pallet based on [XCMP](https://wiki.polkadot.network/docs/learn-crosschain), called [xStorage](https://github.com/crustio/crust/tree/parachain/shadow/crust-collator/pallets/xstorage).
```

> 

Reviewer: ```suggestion
```

> ![xcmp-xstorage](../assets/build-3.png)

Reviewer: ```suggestion
```

Reviewer: Because Crust network is not yet a parachain (aka. parachain candidate), providing a diagram like this on the Polkadot Wiki should be put on hold.

Author: Deleted, will put it after we got polkadot parachain slot

> 

Reviewer: ```suggestion
```

> Crust also supports the majority contract platforms, including Ethereum/Solana/Near/Elrond/... with [cross-chain solution](https://wiki.crust.network/docs/en/buildCrossChainSolution).

Reviewer: ```suggestion
The protocol also supports most smart contract platforms, including Ethereum, with its [cross-chain dStorage solution](https://wiki.crust.network/docs/en/buildCrossChainSolution).
```

> > To learn more about Crust, check its [Decentralized Storage Market](https://wiki.crust.network/docs/en/DSM) and [Guaranteed Proof of Stake](https://wiki.crust.network/docs/en/GPoS)

Reviewer: ```suggestion
> To learn more about Crust, check out the [Crust Network Wiki](https://wiki.crust.network/en).
```

> > Also, you can start with [Build-101](https://wiki.crust.network/docs/en/build101) to start integrating with Crust.

Reviewer: ```suggestion
> Try integrating with Crust by following their [Crust Storage 101](https://wiki.crust.network/docs/en/build101) guide.
```

These layered APIs act as runtime storage that allows you to store data in your blockchain. More information can 
be found at the [`Runtime Storage Page`](https://docs.substrate.io/v3/runtime/storage/) on Substrate Developer Hub.

### Crust Storage

> 

Reviewer: Still needs pretty: https://github.com/w3f/polkadot-wiki#formatting

Reviewer: @badkk still need to format the section so it is consistent with the style guide: 
![image](https://user-images.githubusercontent.com/25497083/151547067-5c469217-7d5d-4745-bfc0-1c56137b653f.png)

[Crust Network](https://crust.network) provides a Web3.0 decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy, and assurance. Crust supports multiple storage-layer protocols such as [IPFS](#ipfs-interplanetary-file-system), and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.

> Crust provides a native cross-chain communnication communication pallet based on [XCMP](https://wiki.polkadot.network/docs/learn-crosschain), called [xStorage](https://github.com/crustio/crust/tree/parachain/shadow/crust-collator/pallets/xstorage). The basic idea of xStorage shows below:

Reviewer: We need to remove "The basic idea of xStorage shows below:" and "build-3.png"
